int-set/src/sparse_bit_set.rs

```diff
@@ -361,13 +452,72 @@ mod test {
         assert_eq!(actual_bytes, expected_bytes);
     }
 
+    #[test]
+    fn generate_spec_example_4() {
+        // Test of reproducing the encoding of example 3 given
+        // in the specification. See:
+        // <https://w3c.github.io/IFT/Overview.html#sparse-bit-set-decoding>
+
+        let actual_bytes = to_sparse_bit_set_with_bf::<4>(&(0..=17).collect());
+        let expected_bytes = [0b00001101, 0b0000_0011, 0b0011_0001];
+
+        assert_eq!(actual_bytes, expected_bytes);
+    }
+
     #[test]
     fn encode_one_level() {
         let actual_bytes = to_sparse_bit_set_with_bf::<8>(&[2, 6].iter().copied().collect());
         let expected_bytes = [0b0_00001_10, 0b01000100];
         assert_eq!(actual_bytes, expected_bytes);
     }
 
+    #[test]
+    fn encode_one_level_filled() {
+        let actual_bytes = to_sparse_bit_set_with_bf::<8>(&(0..=7).collect());
+        let expected_bytes = [0b0_00001_10, 0b00000000];
+        assert_eq!(actual_bytes, expected_bytes);
+    }
+
+    #[test]
+    fn encode_two_level_filled() {
+        let actual_bytes = to_sparse_bit_set_with_bf::<8>(&(3..=21).collect());
+        let expected_bytes = [0b0_00010_10, 0b00000111, 0b11111000, 0b00000000, 0b00111111];
+        assert_eq!(actual_bytes, expected_bytes);
+    }
+
+    #[test]
+    fn encode_two_level_not_filled() {
+        let actual_bytes = to_sparse_bit_set_with_bf::<4>(&[0, 4, 8, 12].iter().copied().collect());
```

Comment:
totally academic but you can do `into_iter()` instead of `iter().copied()`, here.

Reply:
Sounds good, sending another set PR later I'll fix it there.